daml/Utility/Sequence.daml

```diff
@@ -37,6 +37,33 @@ getNextSequenceValue owner name =
         return 0
 
 
+template SequenceGenerator
+  with
+    owner : Party
+    prefix : Text
+    offset: Int
+  where
+    signatory owner
+
+    key (owner, prefix) : (Party, Text)
+    maintainer key._1
+
+    controller owner can
+        SequenceGenerator_BulkCreate : [ContractId Sequence]
```

Comment:
Might be better to return `(SequenceGenerator, [ContractId Sequence])` since you are also updating this contract.